http_parser.c

```diff
@@ -2303,6 +2303,10 @@ http_parser_parse_url(const char *buf, size_t buflen, int is_connect,
   enum http_parser_url_fields uf, old_uf;
   int found_at = 0;
 
+  if (buf == NULL || buf[0] == '\0') {
```

Comment:
This should be just `if (buflen == 0)`.

The NULL check isn't necessary because `buf` isn't allowed to be NULL when `buflen > 0`, and `buf[0]` is an out-of-bound read when `buflen == 0`.

(It also assumes that `buf` is zero-terminated, which isn't necessarily true.)

Reply:
I was thinking that at first, but I realize as `buflen` is based on `buf`, I should pay more attention to that variable.

In case it was passed was below:
```c
http_parser_parse_url("something", 0, 0, <parser>);
```

This means that this should've be done by checking for `buflen  == 0`? as it will be out of bounds immediately as you said.

If I update the code to:

```c
if (buflen == 0) {
    return 1;
}
```

Could it be merged?

Thanks @bnoordhuis for the info.

Reply:
I made the change locally and the tests passed.

From:

```c
if (buf == NULL || buf[0] == '\0') {
  return 1;
}
```
To:

```c
if (buflen == 0) {
    return 1;
}
```

It makes more sense.